### PR TITLE
SCC transformation bug fixes

### PR DESCRIPTION
**Pull request by awnawab**
Small bug fixes for the SCC transformation to fix issues #20, #21 and #23.

### CODE REVIEW

File: transformations/tests/test_single_column_coalesced.py [hunk 1]
```diff
@@ -765,3 +843,127 @@ def test_single_column_coalesced_variable_demotion(frontend, horizontal, vertica
...
+        work(start:end) = 1.
+      case(2)
+        do jl = start,end
+           work(jl) = work(jl) + 2.
+        enddo
+      case(3)
+        do jl = start,end
+           work(jl) = work(jl) + 3.
+        enddo
+        call some_kernel(start, end, work)
```
Comment (reuterbal) [inline transformations/tests/test_single_column_coalesced.py:874]: adding some `default` case here would make sure that's tested as well

**Review by reuterbal — APPROVED** [review]
Very nice additions, neatly done, and excellent testing. Many thanks! Only two minor remarks but no showstoppers!

File: transformations/transformations/single_column_coalesced.py [hunk 2]
```diff
@@ -616,6 +629,13 @@ def process_driver(self, routine, targets=None):
                         p_content = f'parallel loop gang{private_clause}'
                         loop._update(pragma=ir.Pragma(keyword='acc', content=p_content))
                         loop._update(pragma_post=ir.Pragma(keyword='acc', content='end parallel loop'))
+                    # add acc parallel loop gang if the only existing pragma is acc data
+                    elif len(loop.pragma) == 1:
+                        if loop.pragma[0].keyword == 'acc' and 'data' in loop.pragma[0].content:
```
[the remark this answers is not in the document]
Comment (awnawab) [reply transformations/transformations/single_column_coalesced.py]: Ah yes, great catch!

**Comment by reuterbal** [comment]
Thanks, looks great now. I'll let @mlange05 confirm the SCC changes are safe before merging.

**Review by mlange05 — APPROVED** [review]
Looks fantastic and confirmed working with latest EC-physics branches. Excellent test coverage, much appreciated! GTG from me.

**Pull request merged**